client/web-sveltekit/src/lib/navigation/GlobalHeader.svelte

```diff
@@ -0,0 +1,207 @@
+<script context="module" lang="ts">
+    import { type Writable, writable } from 'svelte/store'
+
+    export enum NavigationMode {
+        PlainNavigation = 'global',
+        WithCustomContent = 'with-custom-content',
+    }
+
+    export const navigationModeStore = writable<NavigationMode | `${NavigationMode}`>(NavigationMode.PlainNavigation)
+    export const navigationExtensionElement: Writable<HTMLElement | null> = writable(null)
+</script>
+
+<script lang="ts">
+    import { mdiMenu } from '@mdi/js'
+    import { browser } from '$app/environment'
+
+    import { Badge, Button } from '$lib/wildcard'
+    import { mark } from '$lib/images'
+    import Icon from '$lib/Icon.svelte'
+    import Popover from '$lib/Popover.svelte'
+    import { mainNavigation } from '$lib/navigation/mainNavigation'
+    import GlobalSidebarNavigation from '$lib/navigation/GlobalSidebarNavigation.svelte'
+    import MainNavigationEntry from '$lib/navigation/MainNavigationEntry.svelte'
+
+    import UserMenu from './UserMenu.svelte'
+    import { GlobalNavigation_User } from './GlobalNavigation.gql'
+
+    export let authenticatedUser: GlobalNavigation_User | null | undefined
+    export let handleOptOut: (() => Promise<void>) | undefined
+
+    let isSidebarNavigationOpen: boolean = false
+
+    function setSideNavigationState(open: boolean): void {
+        isSidebarNavigationOpen = open
+    }
+
+    const isDevOrS2 =
+        (browser && window.location.hostname === 'localhost') ||
+        window.location.hostname === 'sourcegraph.sourcegraph.com'
+
+    $: withCustomContent = $navigationModeStore === NavigationMode.WithCustomContent
+</script>
+
+<header class="root">
+    {#if isSidebarNavigationOpen}
+        <GlobalSidebarNavigation onClose={() => setSideNavigationState(false)} />
```

Comment:
Nit: You could just use

```suggestion
        <GlobalSidebarNavigation onClose={() => isSidebarNavigationOpen = false} />
```
and remove the additional function declaration.